sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -700,20 +701,33 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
 
   test("create temporary view using") {
     val csvFile =
-      Thread.currentThread().getContextClassLoader.getResource("test-data/cars.csv").toString
-    withView("testview") {
-      sql(s"CREATE OR REPLACE TEMPORARY VIEW testview (c1 String, c2 String)  USING " +
-        "org.apache.spark.sql.execution.datasources.csv.CSVFileFormat  " +
-        s"OPTIONS (PATH '$csvFile')")
+      Thread.currentThread().getContextClassLoader.getResourceAsStream("test-data/cars.csv")
```

Comment:
But isn't the resource you're retrieving to copy still the 'wrong' resource, because you accessed it by a relative path?
If you say it definitely resolve the problem because you've tested it I do believe it, but I'm not clear why it differs.